backend/endpoints/scan.py

```diff
@@ -86,7 +104,106 @@ async def scan_platforms(
+        for fs_emulator, fs_save_filename in fs_assets["saves"]:
+            scanned_save = scan_save(
+                platform=scanned_platform,
+                file_name=fs_save_filename,
+                emulator=fs_emulator,
+            )
+
+            save = dbh.get_save_by_filename(scanned_platform.slug, fs_save_filename)
+            if save:
+                dbh.update_save(
+                    save.id, {"file_size_bytes": scanned_save.file_size_bytes}
+                )
+                continue
+
+            scanned_save.emulator = fs_emulator
+            scanned_save.platform_slug = scanned_platform.slug
+
+            rom = dbh.get_rom_by_filename_no_tags(scanned_save.file_name_no_tags)
+            if rom:
+                scanned_save.rom_id = rom.id
+                dbh.add_save(scanned_save)
+
+        # Scanning states
+        log.info(f"\t · {len(fs_assets['states'])} states found")
+        for fs_emulator, fs_state_filename in fs_assets["states"]:
+            scanned_state = scan_state(
+                platform=scanned_platform,
+                emulator=fs_emulator,
+                file_name=fs_state_filename,
+            )
+
+            state = dbh.get_state_by_filename(scanned_platform.slug, fs_state_filename)
+            if state:
+                dbh.update_state(
+                    state.id, {"file_size_bytes": scanned_state.file_size_bytes}
+                )
+                continue
+
+            scanned_state.emulator = fs_emulator
+            scanned_state.platform_slug = scanned_platform.slug
+
+            rom = dbh.get_rom_by_filename_no_tags(scanned_state.file_name_no_tags)
+            if rom:
+                scanned_state.rom_id = rom.id
+                dbh.add_state(scanned_state)
+
+        # Scanning screenshots
+        log.info(f"\t · {len(fs_assets['screenshots'])} screenshots found")
+        for fs_screenshot_filename in fs_assets["screenshots"]:
+            scanned_screenshot = scan_screenshot(
+                file_name=fs_screenshot_filename, fs_platform=scanned_platform.slug
+            )
+
+            screenshot = dbh.get_screenshot_by_filename(fs_screenshot_filename)
+            if screenshot:
+                dbh.update_screenshot(
+                    screenshot.id,
+                    {"file_size_bytes": scanned_screenshot.file_size_bytes},
+                )
+                continue
+
+            scanned_screenshot.platform_slug = scanned_platform.slug
+
+            rom = dbh.get_rom_by_filename_no_tags(scanned_screenshot.file_name_no_tags)
+            if rom:
+                scanned_screenshot.rom_id = rom.id
+                dbh.add_screenshot(scanned_screenshot)
+
+        dbh.purge_saves(scanned_platform.slug, [s for _e, s in fs_assets["saves"]])
+        dbh.purge_states(scanned_platform.slug, [s for _e, s in fs_assets["states"]])
+        dbh.purge_screenshots(fs_assets["screenshots"], scanned_platform.slug)
```

Comment:
All of these are files nested under platforms, and for saves/states with optional emulator folders